### PR TITLE
Fix Doxygen: BTD

### PR DESCRIPTION
**Pull request by ax3l**
Fix doxygen errors & warnings.
Related to #2482.

### CODE REVIEW

File: Source/Diagnostics/BTD_Plotfile_Header_Impl.H [hunk 1]
```diff
@@ -74,25 +74,25 @@ public:
     void WriteHeader ();
 
     /** Sets the physical simulation time, m_time, in the Header file to a new_time.
-     *  \param[in] Real new_time
+     *  \param[in] new_time ...?
```
Comment (ax3l) [inline Source/Diagnostics/BTD_Plotfile_Header_Impl.H]: Please add a suggestion during review :)

File: Source/Diagnostics/BTD_Plotfile_Header_Impl.H [hunk 2]
```diff
@@ -74,25 +74,25 @@ public:
...
      **/
     void set_time ( amrex::Real new_time) { m_time = new_time;}
     /** Sets the timestep, m_timestep, in the Header file to a new_timestep
-     *  \param[in] int new_timestep
+     *  \param[in] new_timestep ...?
```
Comment (ax3l) [inline Source/Diagnostics/BTD_Plotfile_Header_Impl.H]: Please add a suggestion during review :)

File: Source/Diagnostics/ComputeDiagFunctors/BackTransformFunctor.H [hunk 3]
```diff
@@ -57,14 +60,16 @@ public:
...
-     * \param[in] buffer_box, Box with index-space in lab-frame for the ith buffer
-     * \param[in] k_index_zlab, k-index in the lab-frame corresponding to the
-     *                          current z co-ordinate in the lab-frame for the ith buffer.
-     * \param[in] max_box_size, maximum box size for the multifab to generate box arrays
+     * \param[in] i_buffer        index of the snapshot
+     * \param[in] ZSliceInDomain  if the z-slice at current_z_boost is within
+     *                            the boosted-frame and lab-frame domain.
+     *                            The fields are sliced and back-transformed only
+     *                            if this value is true.
+     * \param[in] current_z_boost ...?
```
Comment (ax3l) [inline Source/Diagnostics/ComputeDiagFunctors/BackTransformFunctor.H]: Please add a suggestion during review :)

File: Source/Diagnostics/ComputeDiagFunctors/BackTransformFunctor.H [hunk 4]
```diff
@@ -32,8 +32,11 @@ public:
      * \param[in] mf_src cell-centered multifab containing all user-requested
                          fields in boosted-frame
      * \param[in] lev level of multifab.
-     * \param[in] ncom number of components of mf_src to Lorentz-Transform
-                       and store in destination multifab.
+     * \param[in] ncomp number of components of mf_src to Lorentz-Transform
+                        and store in destination multifab.
+     * \param[in] num_buffers ...?
```
Comment (ax3l) [inline Source/Diagnostics/ComputeDiagFunctors/BackTransformFunctor.H]: Please add a suggestion during review :)

File: Source/Diagnostics/ComputeDiagFunctors/BackTransformFunctor.H [hunk 5]
```diff
@@ -32,8 +32,11 @@ public:
...
+     * \param[in] varnames ...?
```
Comment (ax3l) [inline Source/Diagnostics/ComputeDiagFunctors/BackTransformFunctor.H]: Please add a suggestion during review :)

File: Source/Diagnostics/BTDiagnostics.H [hunk 6]
```diff
@@ -43,6 +43,7 @@ private:
      *  The data is flushed when the buffer is full and/or
      *  when the simulation ends or when forced.
      * \param[in] step current time step
+     * \param[in] i_buffer ...?
```
Comment (ax3l) [inline Source/Diagnostics/BTDiagnostics.H]: Please add a suggestion during review :)

File: Source/Diagnostics/BTD_Plotfile_Header_Impl.H (hunk 1 above)
Comment (RevathiJambunathan) [reply Source/Diagnostics/BTD_Plotfile_Header_Impl.H]:
```suggestion
     *  \param[in] new_time current time in the boosted-frame read from Header plotfile
```

File: Source/Diagnostics/BTD_Plotfile_Header_Impl.H (hunk 2 above)
Comment (RevathiJambunathan) [reply Source/Diagnostics/BTD_Plotfile_Header_Impl.H]:
```suggestion
     *  \param[in] new_timestep current timestep in the boosted-frame (read from the Header file of the newly flushed buffer)
```

File: Source/Diagnostics/BTDiagnostics.H [hunk 7]
```diff
@@ -203,45 +205,46 @@ private:
      *  snapshot, i_buffer, at level, lev
      *
      * \param[in] i_buffer buffer-id of the back-transformed snapshot
-     * \param[in] lev, level at which the output buffer MultiFab is defined
+     * \param[in] lev      level at which the output buffer MultiFab is defined
```
Comment (RevathiJambunathan) [inline Source/Diagnostics/BTDiagnostics.H]: is there a reason you chose to keep this as level, but the definition above as mesh-refinement level?

File: Source/Diagnostics/ComputeDiagFunctors/BackTransformFunctor.H (hunk 4 above)
Comment (RevathiJambunathan) [inline Source/Diagnostics/ComputeDiagFunctors/BackTransformFunctor.H]:
```suggestion
     * \param[in] num_buffers number of user-defined snapshots in the back-transformed lab-frame
```

File: Source/Diagnostics/ComputeDiagFunctors/BackTransformFunctor.H (hunk 5 above)
Comment (RevathiJambunathan) [inline Source/Diagnostics/ComputeDiagFunctors/BackTransformFunctor.H]:
```suggestion
     * \param[in] varnames names of the field-components as defined by the user for back-transformed diagnostics.
```

File: Source/Diagnostics/ComputeDiagFunctors/BackTransformFunctor.H (hunk 3 above)
Comment (RevathiJambunathan) [reply Source/Diagnostics/ComputeDiagFunctors/BackTransformFunctor.H]:
```suggestion
     * \param[in] current_z_boost current z-coordinate of the slice in boosted-frame
```

**Review by RevathiJambunathan — COMMENTED** [review]
Thank you for this PR, Axel

File: Source/Diagnostics/BTDiagnostics.H (hunk 7 above)
Comment (ax3l) [reply Source/Diagnostics/BTDiagnostics.H]: `lev` is the argument name below and the rest is the description (after the first space) :)

File: Source/Diagnostics/BTDiagnostics.H (hunk 6 above)
Comment (RevathiJambunathan) [reply Source/Diagnostics/BTDiagnostics.H]:
```suggestion
     * \param[in] i_buffer snapshot index
```

File: Source/Diagnostics/BTDiagnostics.H (hunk 7 above)
Comment (RevathiJambunathan) [reply Source/Diagnostics/BTDiagnostics.H]: I mean -- on line 95 you have `lev` mesh-refinement level
and here on line 208, `lev` level

File: Source/Diagnostics/BTDiagnostics.H [hunk 8]
```diff
@@ -262,13 +265,13 @@ private:
...
     /** Reset buffer counter to zero.
-     * \param[in] i_buffer, ith buffer for which the counter is set to zero.
+     * \param[in] i_buffer ith buffer for which the counter is set to zero.
      */
     void ResetBufferCounter(int i_buffer) {
         m_buffer_counter[i_buffer] = 0;
     }
     /** Increment buffer counter by one when the buffer of a snapshot has been flushed.
-     * \param[in] i_buffer, ith buffer for which the counter is incremented.
+     * \param[in] i_buffer ith buffer for which the counter is incremented.
```
Comment (RevathiJambunathan) [inline Source/Diagnostics/BTDiagnostics.H]:
```suggestion
     * \param[in] i_buffer snapshot index for which the counter is incremented.
```

File: Source/Diagnostics/BTDiagnostics.H [hunk 9]
```diff
@@ -262,13 +265,13 @@ private:
     }
 
     /** Reset buffer counter to zero.
-     * \param[in] i_buffer, ith buffer for which the counter is set to zero.
+     * \param[in] i_buffer ith buffer for which the counter is set to zero.
```
Comment (RevathiJambunathan) [inline Source/Diagnostics/BTDiagnostics.H]:
```suggestion
     * \param[in] i_buffer snapshot index for which the counter is set to zero.
```

File: Source/Diagnostics/BTDiagnostics.H [hunk 10]
```diff
@@ -203,45 +205,46 @@ private:
...
+     *  \param[in] ref_ratio  refinement ratio in the z-direction at level, lev-1.
+     *             The ref-ratio in the z-direction for single-level diagnostics is 1.
+     */
     amrex::Real dz_lab (amrex::Real dt, amrex::Real ref_ratio);
     /** Compute k-index corresponding to current lab-frame z co-ordinate (m_current_z_lab)
-     *  for the ith buffer, i_buffer, and at level, lev.
-     * \param[in] i_buffer, ith buffer
-     * \param[in] lev, level at which the lab-frame z-index is computed
+     *  for the ith buffer i_buffer, and at level, lev.
+     * \param[in] i_buffer ith buffer
```
Comment (RevathiJambunathan) [inline Source/Diagnostics/BTDiagnostics.H]:
```suggestion
     * \param[in] i_buffer snapshot index
```

File: Source/Diagnostics/BTDiagnostics.H (hunk 7 above)
Comment (ax3l) [reply Source/Diagnostics/BTDiagnostics.H]: Oh, yes I added to the description so document "lev" as more than "level" :D I will update here, too.
```suggestion
     * \param[in] lev      mesh-refinement level at which the output buffer MultiFab is defined
```

File: Source/Diagnostics/BTDiagnostics.H [hunk 11]
```diff
@@ -203,45 +205,46 @@ private:
...
+     * \param[in] lev      level at which the lab-frame z-index is computed
```
Comment (ax3l) [inline Source/Diagnostics/BTDiagnostics.H]:
```suggestion
     * \param[in] lev      mesh-refinement level at which the lab-frame z-index is computed
```

File: Source/Diagnostics/ComputeDiagFunctors/BackTransformFunctor.H [hunk 12]
```diff
@@ -32,8 +32,11 @@ public:
      * \param[in] mf_src cell-centered multifab containing all user-requested
                          fields in boosted-frame
      * \param[in] lev level of multifab.
```
Comment (ax3l) [inline Source/Diagnostics/ComputeDiagFunctors/BackTransformFunctor.H]:
```suggestion
     * \param[in] lev mesh-refinement level of multifab.
```

**Review by RevathiJambunathan — APPROVED** [review]

**Pull request merged**